com.unity.ml-agents.extensions/Runtime/Sensors/GridSensorBase.cs

```diff
@@ -0,0 +1,336 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using Unity.MLAgents.Sensors;
+using UnityEngine.Profiling;
+using Object = UnityEngine.Object;
+
+[assembly: InternalsVisibleTo("Unity.ML-Agents.Extensions.TestUtils")]
+namespace Unity.MLAgents.Extensions.Sensors
+{
+    /// <summary>
+    /// Grid-based sensor.
+    /// </summary>
+    public class GridSensorBase : ISensor, IBuiltInSensor, IDisposable
+    {
+        string m_Name;
+        Vector3 m_CellScale;
+        Vector3Int m_GridSize;
+        int[] m_ChannelDepths;
```

Comment:
Are m_ChannelDepths still needed? I can see how a custom implementation might need them, but not the ones you have now. Seems like it would eliminate some of the confusing parts of the interface.

Reply:
yes I noticed that too and I removed it